### PR TITLE
Timur attribute ordering

### PR DESCRIPTION
**Pull request by coleshaw**
This PR addresses mountetna/timur#277. This enforces the same attribute ordering that we're using in the model map page, extracts that code and puts it into a shared utility that the Browser Pane uses ... @graft , this shouldn't override any manually defined views, right, but it will sort any attributes that are passed in from that view? Perhaps that is not desired before and the sorting needs to happen higher up in the component tree...I will do some more testing and play around with it.

This works currently for default views. Not sure how it interacts with manually defined views, and I need to test that further.

### CODE REVIEW

File: timur/lib/client/jsx/utils/attributes.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    identifier: 2,
+    collection: 3,
+    table: 4,
+    child: 5,
+    link: 6,
+    file: 7,
+    image: 8
+  }[attribute_type] || 9);
+
+export const sortAttributes = (attributes) =>
```
Comment (corps) [inline timur/lib/client/jsx/utils/attributes.js:13]: Could we use this function [here](https://github.com/mountetna/monoetna/blob/master/timur/lib/client/jsx/selectors/search.js#L78) as well, for even mooore consistency?

**Review by corps — APPROVED** [review]

**Comment by graft** [comment]
Hmm, I am a bit confused by this since the ViewAttribute includes an explicit `index_order`, but this is apparently unused in BrowserPane, which just seems to use Object.keys(attributes), i.e., no order at all. Calling `sortAttributes` instead therefore doesn't seem like it could break anything, but the existing behavior doesn't seem to make use of the available `index_order` in the ViewAttribute model, as it ought to, and now `sortAttributes` takes the place of such ordering.

**Comment by coleshaw** [comment]
> Hmm, I am a bit confused by this since the ViewAttribute includes an explicit `index_order`, but this is apparently unused in BrowserPane, which just seems to use Object.keys(attributes), i.e., no order at all. Calling `sortAttributes` instead therefore doesn't seem like it could break anything, but the existing behavior doesn't seem to make use of the available `index_order` in the ViewAttribute model, as it ought to, and now `sortAttributes` takes the place of such ordering.

Hm, so that's interesting. On a default view, with no custom view defined, I don't see any `index_order` on the list of pane attributes.  I also don't see how to get `index_order` [from the template](https://github.com/mountetna/timur/blob/master/lib/client/jsx/selectors/tab_selector.jsx#L62), since that information isn't present in the template. So somehow `sortAttributes` or equivalent would have to know if a defined View were being used and stick to the order defined in the view, or set a default ordering otherwise (otherwise it may re-order something defined by ViewAttribute). Did I understand that right?

Let me see if I can move the method call to the selector, maybe that makes more sense.

**Comment by coleshaw** [comment]
I think the ViewAttribute `index_order` is implied in the order of the Hash keys returned by the API....so it just happens to work out because of how the backend returns things when the view is defined. But BrowserPane doesn't reinforce or check that the order matches `index_order`. I've updated BrowserPane to respect the `index_order`.

**Pull request merged**